sdk/keyvault/azure-security-keyvault-keys/CMakeLists.txt

```diff
@@ -138,6 +138,8 @@ create_code_coverage(keyvault azure-security-keyvault-keys azure-security-keyvau
 get_az_version("${CMAKE_CURRENT_SOURCE_DIR}/src/private/package_version.hpp")
 generate_documentation(azure-security-keyvault-keys ${AZ_LIBRARY_VERSION})
 
+add_subdirectory(test/macros-build)
+
```

Comment:
I imagine this should only be built if `BUILD_TESTING` is true. Move inside the if?

Reply:
This test directory is still being add outside of when `BUILD_TESTING` is true.

Reply:
yes, it's a build test, i argue it needs to always be tested , especially since we have magic flags in the build to check to make deprecated warnings as errors

Reply:
> yes, it's a build test, i argue it needs to always be tested

Agreed. We should be building/testing this to confirm the fix. All our tests run with `BUILD_TESTING` turned on/in CI.

For maintainability, I recommend adding anything related to the `test` directory only when `BUILD_TESTING` is true. Otherwise, if we don't keep that invariant true, we could see unintended CI failures.

Also, **more importantly**, I don't think we should be shipping this macros-build app along with the test directory to our end customers, as part of the product.

Can you confirm if we package this up and ship the SDK to customers, when they install the package, they don't get this test app on disk? If they do, we **definitely** need to move this to the `BUILD_TESTING` if check a few lines below in the CMakeLists.txt. We don't want to ship this to customers, right?

Any reason, that I might not be aware, that would prevent us from moving this `add_subdirectory` in that if-block? To be clear, in here:
https://github.com/Azure/azure-sdk-for-cpp/blob/7d7946110219f9b6d2971cddb444973988ca8d15/sdk/keyvault/azure-security-keyvault-keys/CMakeLists.txt#L143-L152

Reply:
i moved it